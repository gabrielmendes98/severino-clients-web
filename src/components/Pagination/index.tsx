import MuiPagination, { PaginationProps } from '@mui/material/Pagination';
import { DEFAULT_LIST_COUNT } from 'common/constants';

interface Props extends PaginationProps {
  total?: number;
  perPage?: number;
}

const Pagination = ({
  color = 'primary',
  total = 0,
  perPage = DEFAULT_LIST_COUNT,
  ...props
}: Props) => (
  <MuiPagination color={color} count={Math.ceil(total / perPage)} {...props} />
);

export default Pagination;
