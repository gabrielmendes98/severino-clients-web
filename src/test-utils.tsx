import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
