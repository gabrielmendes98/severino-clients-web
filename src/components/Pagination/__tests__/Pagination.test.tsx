import { render } from 'test-utils';
import Pagination from '../index';

it('should match snapshot', () => {
  const { container } = render(<Pagination total={10} perPage={3} />);
  expect(container).toMatchSnapshot();
});
