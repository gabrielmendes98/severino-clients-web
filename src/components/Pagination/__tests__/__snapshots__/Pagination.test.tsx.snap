// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
  >
    <ul
      class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1qrkoh3-MuiButtonBase-root-MuiPaginationItem-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateBeforeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1qrkoh3-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1qrkoh3-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1qrkoh3-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          3
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-page css-1qrkoh3-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          4
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary MuiPaginationItem-previousNext css-1qrkoh3-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="NavigateNextIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
